web/src/api/index.ts

```diff
@@ -1,7 +1,37 @@
 import createClient from 'openapi-fetch';
 import { components, paths } from './schema';
+import { createLocalStorageSignal } from '../utils/localstorage';
 
 const client = createClient<paths>({ baseUrl: '/api' });
+const [currentSessionToken, setCurrentSessionToken] =
+  createLocalStorageSignal('sessionToken');
+export const isLoggedIn = () => currentSessionToken() !== undefined;
+function sessionOptions() {
+  const sessionToken = currentSessionToken();
+  const headers: HeadersInit = {};
+  if (sessionToken) {
+    headers.Authorization = `Bearer ${sessionToken}`;
+  }
+  return { headers };
+}
+// Try to validate that the initial session token is still valid, and log the user out otherwise
+{
+  const initialSessionToken = currentSessionToken();
+  if (initialSessionToken !== undefined) {
+    client
+      .get('/ping', sessionOptions())
+      .then((pingResponse) => {
+        if (pingResponse.response.status === 401) {
+          setCurrentSessionToken();
+        }
+      })
+      // We don't want to block page loads on waiting for this validation
+      // eslint-disable-next-line unicorn/prefer-top-level-await
+      .catch((error) =>
+        console.error('Failed to validate session token', error),
+      );
+  }
+}
```

Comment:
I'm not a super huge fan of this being executed automagically. We might want to move this to an `initAuth` function, which gets called when the page loads.

Reply:
436880001989c8518af7ffa78c38b61eaf2a7a9d